tests/test_context.py

```diff
@@ -323,5 +323,5 @@ def cli(ctx, option):
 
 
 def test_validate_parameter_source():
-    with pytest.raises(ValueError):
-        ParameterSource.validate("NOT_A_VALID_PARAMETER_SOURCE")
+    with pytest.raises(KeyError):
+        ParameterSource["NOT_A_VALID_PARAMETER_SOURCE"]
```

Comment:
Probably this test can go away. Testing basic stdlib functionality doesn't make much sense IMO

Reply:
I agree, I wasn't sure if I should remove the test but I'm happy to do so after your suggestion